src/settings/groups/GroupDetailsView.ts

```diff
@@ -0,0 +1,276 @@
+import {formatDateWithMonth, formatStorageSize} from "../../misc/Formatter.js"
+import {lang} from "../../misc/LanguageViewModel.js"
+import {getFirstOrThrow, neverNull} from "@tutao/tutanota-utils"
+import {GroupType} from "../../api/common/TutanotaConstants.js"
+import type {TableAttrs} from "../../gui/base/Table.js"
+import {ColumnWidth, Table, TableLineAttrs} from "../../gui/base/Table.js"
+import {logins} from "../../api/main/LoginController.js"
+import {Icons} from "../../gui/base/icons/Icons.js"
+import {showProgressDialog} from "../../gui/dialogs/ProgressDialog.js"
+import type {EntityUpdateData} from "../../api/main/EventController.js"
+import {TextField} from "../../gui/base/TextField.js"
+import type {DropDownSelectorAttrs} from "../../gui/base/DropDownSelector.js"
+import {DropDownSelector} from "../../gui/base/DropDownSelector.js"
+import type {UpdatableSettingsDetailsViewer} from "../SettingsView.js"
+import {locator} from "../../api/main/MainLocator.js"
+import {assertMainOrNode} from "../../api/common/Env.js"
+import {IconButton, IconButtonAttrs} from "../../gui/base/IconButton.js"
+import {ButtonSize} from "../../gui/base/ButtonSize.js"
+import {GroupDetailsModel} from "./GroupDetailsModel.js"
+
+assertMainOrNode()
+
+export class GroupDetailsView implements UpdatableSettingsDetailsViewer {
+	constructor(
+		private readonly model: GroupDetailsModel,
+	) {
+	}
+
+	/**
+	 * render the header that tells us what type of group we have here
+	 * @private
+	 */
+	private renderHeader(): Children {
+		return m(".h4.mt-l", getGroupTypeDisplayName(this.model.getGroupType()))
+	}
+
+	renderView(): Children {
+		return m("#user-viewer.fill-absolute.scroll.plr-l", [
+			this.renderHeader(),
+			...this.renderCommonInfo(),
+			...this.renderTypeDependentInfo()
+		])
+	}
+
+	private renderTypeDependentInfo(): ChildArray {
+		return this.model.isMailGroup()
+			? this.renderMailGroupInfo()
+			: this.renderLocalAdminGroupInfo()
+	}
+
+	/**
+	 * render the fields that are common to all group types
+	 * @private
+	 */
+	private renderCommonInfo(): ChildArray {
+		return [
+			this.renderCreatedTextField(),
+			this.renderNameField(),
+			this.renderAdministratedByDropdown(),
+			this.renderStatusSelector(),
+			this.renderMembersTable(),
+		]
+	}
+
+	private renderCreatedTextField(): Children {
+		return m(TextField, {label: "created_label", value: formatDateWithMonth(this.model.getCreationDate()), disabled: true})
+	}
+
+	private renderAdministratedByDropdown(): Children {
+		if (!logins.getUserController().isGlobalAdmin()) return null
+		const administratedByInfo = this.model.createAdministratedByInfo()
+		if (!administratedByInfo) return null
+		const {options, currentVal} = administratedByInfo
+		return m(DropDownSelector, {
+			label: "administratedBy_label",
+			items: options,
+			selectedValue: currentVal,
+			selectionChangedHandler: id => this.model.onAdministratedBySelected(id),
+		} as DropDownSelectorAttrs<Id | null>)
```

Comment:
casting is not nice, would be good to use `satisfies` but we need to update ts for that